diffsync/__init__.py

```diff
@@ -132,27 +132,27 @@ def __init_subclass__(cls):
         if attr_child_overlap:
             raise AttributeError(f"Fields {attr_child_overlap} are included in both _attributes and _children.")
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return f'{self.get_type()} "{self.get_unique_id()}"'
 
-    def __str__(self):
+    def __str__(self) -> str:
         return self.get_unique_id()
 
-    def dict(self, **kwargs) -> dict:
+    def dict(self, **kwargs: Any) -> Dict:
```

Comment:
For `**kwargs` you need to annotate the value types as per [PEP 484](https://peps.python.org/pep-0484/#arbitrary-argument-lists-and-default-argument-values). I went the lazy route here and picked `typing.Any` because I didn't know what exactly the values might be. Need to check this again.